packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -416,22 +444,71 @@ export class TokenRatesController extends BaseController<
     if (this.tokenList.length === 0 || this.disabled) {
       return;
     }
-    const slug = await this.getChainSlug();
+
+    const { chainId, nativeCurrency } = this.config;
+    const tokenAddresses = this.tokenList.map((token) => token.address);
+    await this.updateExchangeRatesByChainId({
+      chainId,
+      nativeCurrency,
+      tokenAddresses,
+    });
+
+    this.update({
+      contractExchangeRates:
+        this.state.contractExchangeRatesByChainId[chainId][nativeCurrency],
+    });
+  }
+
+  /**
+   * Updates exchange rates for all tokens.
+   *
+   * @param options - The options to fetch exchange rates.
+   * @param options.chainId - The chain ID.
+   * @param options.nativeCurrency - The ticker for the chain.
+   * @param options.tokenAddresses - The addresses for the token contracts.
+   */
+  async updateExchangeRatesByChainId({
```

Comment:
generally I've been trying to use the original method and adding optional params and branching logic within, but I can see this would have been pretty messy with that approach so I think this is okay.

Reply:
the original `updateExchangeRates` method is still there and its function signature is unchanged. `updateExchangeRatesByChainId` does not replace `updateExchangeRates`

Reply:
now i understand, you're saying that you wouldn't have added a separate method, not that the original method has changed.

Yeah... it's a little bit tricky with this one because of `this.tokenList` and the `this.disabled` check